tests/ZfcRbacTest/Collector/RbacCollectorTest.php

```diff
@@ -176,4 +178,126 @@ public function testCanCollect()
+     */
+    public function testCollectPermissionsMethod()
+    {
+        $expectedCollection = [
+            'guards' => [],
+            'roles'  => ['role-with-permission-method'],
+            'permissions' => [
+                'role-with-permission-method' => ['permission-method-a', 'permission-method-b'],
+            ],
+            'options' => [
+                'guest_role' => 'guest',
+                'protection_policy' => GuardInterface::POLICY_ALLOW,
+            ],
+        ];
+
+        $collection = $this->collectPermissionsPropertyTestBase(new MockRoleWithPermissionMethod());
+        $this->assertEquals($expectedCollection, $collection);
+    }
+
+    /**
+     * Base method for the *collectPermissionProperty tests
+     * @param RoleInterface $role
+     * @return array|\string[]
+     */
+    private function collectPermissionsPropertyTestBase(RoleInterface $role)
+    {
+        $serviceManager = $this->getMock('Zend\ServiceManager\ServiceLocatorInterface');
+
+        $application = $this->getMock('Zend\Mvc\Application', [], [], '', false);
+        $application->expects($this->once())->method('getServiceManager')->will($this->returnValue($serviceManager));
+
+        $mvcEvent = new MvcEvent();
+        $mvcEvent->setApplication($application);
+
+        $identity = $this->getMock('ZfcRbac\Identity\IdentityInterface');
+        $identity->expects($this->once())
+            ->method('getRoles')
+            ->will($this->returnValue([$role]));
+
+        $identityProvider = $this->getMock('ZfcRbac\Identity\IdentityProviderInterface');
+        $identityProvider->expects($this->once())
+            ->method('getIdentity')
+            ->will($this->returnValue($identity));
+
+        $roleProvider = $this->getMock('ZfcRbac\Role\RoleProviderInterface');
+
+        $roleService = new RoleService($identityProvider,
+            $roleProvider,
+            new RecursiveRoleIteratorStrategy());
+
+        $serviceManager->expects($this->at(0))
+            ->method('get')
+            ->with('ZfcRbac\Service\RoleService')
+            ->will($this->returnValue($roleService));
+
+        $serviceManager->expects($this->at(1))
+            ->method('get')
+            ->with('ZfcRbac\Options\ModuleOptions')
+            ->will($this->returnValue(new ModuleOptions()));
+
+        $collector = new RbacCollector();
+        $collector->collect($mvcEvent);
+
+        $collector->unserialize($collector->serialize());
+        return $collector->getCollection();
+    }
+
 }
+
+class MockRoleWithPermissionProperty implements RoleInterface
+{
+    private $permissions = ['permission-property-a', 'permission-property-b'];
+
+    public function getName()
+    {
+        return 'role-with-permission-property';
+    }
+    public function hasPermission($permission)
+    {
+        return false;
+    }
+}
+
+class MockRoleWithPermissionMethod implements RoleInterface
```

Comment:
You should create those two classes in the asset folder: https://github.com/ZF-Commons/zfc-rbac/tree/master/tests/ZfcRbacTest/Asset :)